systemtest/src/test/java/io/strimzi/systemtest/dynamicconfiguration/DynamicConfigurationSharedST.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.dynamicconfiguration;
+
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.resources.ResourceManager;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static io.strimzi.systemtest.Constants.DYNAMIC_CONFIGURATION;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.is;
+
+@Tag(REGRESSION)
+@Tag(DYNAMIC_CONFIGURATION)
+public class DynamicConfigurationSharedST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(DynamicConfigurationSharedST.class);
+    private static final String NAMESPACE = "kafka-configuration-shared-cluster-test";
+
+    @ParameterizedTest
+    @CsvSource({
+        "background.threads, " + 12,
+        "compression.type,  snappy",
+        "compression.type,  gzip",
+        "compression.type,  lz4",
+        "compression.type,  zstd",
+        "log.flush.interval.ms, " + 20,
+        "log.retention.ms,  " + 20,
+        "log.retention.bytes, " + 250,
+        "log.segment.bytes,   " + 1_100,
+        "log.segment.delete.delay.ms,  " + 400,
+        "log.roll.jitter.ms, " + 500,
+        "log.roll.ms, " + 300,
+        "log.cleaner.dedupe.buffer.size, " + 4_000_000,
+        "log.cleaner.delete.retention.ms, " + 1_000,
+        "log.cleaner.io.buffer.load.factor, " + 12,
+        "log.cleaner.io.buffer.size, " + 10_000,
+        "log.cleaner.io.max.bytes.per.second, " + 1.523,
+        "log.cleaner.max.compaction.lag.ms, " + 32_000,
+        "log.cleaner.min.compaction.lag.ms, " + 1_000,
+        "log.preallocate, " + true,
+        "max.connections, " + 10,
+        "max.connections.per.ip, " + 20,
+        "unclean.leader.election.enable, " + true,
+        "message.max.bytes, " + 2048,
```

Comment:
Is this a complete list? If not, how did you decide which to tests? How will we keep it up to date as more configs are added in new versions of Kafka?

It's not a problem for this PR, but I do think we really need, written down, a list of which configs can be changed dynamically and which not. It needs to be part of the docs, so users can have certainty that changing some particular configs will or won't result in a rolling restart. Because new configs get regularly added it needs to be generated from the code, rather than something which we maintain by hand. @stanlyDoge would you be able to do this?

Reply:
It is an almost a complete list of Dyn. configuration. I picked randomly, which were not `read-only` and also which does not have this
```
 FORBIDDEN_PREFIXES = "listeners, advertised., broker., listener., host.name, port, inter.broker.listener.name, sasl., ssl., security., password., principal.builder.class, log.dir, zookeeper.connect, zookeeper.set.acl, authorizer., super.user, cruise.control.metrics.topic, cruise.control.metrics.reporter.bootstrap.servers;
```

Reply:
Hmmm, I can create a .csv files, which will bind to specific version :)

Reply:
No. You could include columns in the CSV defining the version range between which to test with that config, skipping versions outside that range. But the key point here is that to get complete coverage we need to automate this, which means not relying on an annotation to drive the values, but that generated file I described.